internal/juju/deployments.go

```diff
@@ -182,7 +215,73 @@ func (c deploymentsClient) CreateDeployment(input *CreateDeploymentInput) (strin
 		NumUnits:        input.Units,
 		Series:          resultOrigin.Series,
 	})
-	return appName, err
+	return &CreateDeploymentResponse{
+		AppName:  appName,
+		Revision: *origin.Revision,
+		Series:   series,
+	}, err
+}
+
+func (c deploymentsClient) ReadDeployment(input *ReadDeploymentInput) (*ReadDeploymentResponse, error) {
+	conn, err := c.GetConnection(&input.ModelUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	applicationAPIClient := apiapplication.NewClient(conn)
+	defer applicationAPIClient.Close()
+
+	charmsAPIClient := apicharms.NewClient(conn)
+	defer charmsAPIClient.Close()
+
+	clientAPIClient := apiclient.NewClient(conn)
+	defer clientAPIClient.Close()
+
+	apps, err := applicationAPIClient.ApplicationsInfo([]names.ApplicationTag{names.NewApplicationTag(input.AppName)})
+	if err != nil {
+		return nil, err
+	}
+	appInfo := apps[0].Result
+
+	status, err := clientAPIClient.Status(nil)
+	if err != nil {
+		return nil, err
+	}
+	appStatus := status.Applications[input.AppName]
```

Comment:
Is there a guarantee that the `AppName` will be present?

Reply:
Sorry, I wasn't clear. I meant is `status.Applications[input.AppName]` guaranteed to yield the status?